kasa/smart/smartdevice.py

```diff
@@ -197,21 +204,122 @@ async def update(self, update_children: bool = False):
         if not self._features:
             await self._initialize_features()
 
-        _LOGGER.debug("Got an update: %s", self._last_update)
+        _LOGGER.debug(
+            "Update completed %s: %s",
+            self.host,
+            self._last_update if first_update else resp,
+        )
 
     def _handle_module_post_update_hook(self, module: SmartModule) -> bool:
         try:
             module._post_update_hook()
             return True
         except Exception as ex:
-            _LOGGER.error(
+            _LOGGER.warning(
                 "Error processing %s for device %s, module will be unavailable: %s",
                 module.name,
                 self.host,
                 ex,
             )
             return False
 
+    async def _modular_update(
+        self, first_update: bool, update_time: float
+    ) -> dict[str, Any]:
+        """Update the device with via the module queries."""
+        updated_modules: set[type[SmartModule]] = set()
+        if first_update:
+            updated_modules = {DeviceModule, ChildDevice, Cloud}
+
+        req: dict[str, Any] = {}
+        # Keep a track of actual module queries so we can track the time for
+        # modules that do not need to be updated frequently
+        module_queries: list[SmartModule] = []
```

Comment:
An open question, would it make sense to inverse this handling and do it as a part of the smartmodule? I.e., leverage the delay interval to decide what to query? This would allow overriding the behavior on module level by overriding `query()`.

Reply:
Thought about this but decided it's cleaner to logic to decide whether to wait or not in the one place in `_modular_update`.  Simplified that method though so looks nicer.